sunpy/net/dataretriever/sources/noaa.py

```diff
@@ -200,6 +149,22 @@ def _get_url_for_timerange(self, timerange, **kwargs):
             result.append(url)
         return result
 
+    def search(self, *args, **kwargs):
+        extractor1 = '{}/warehouse/{:4d}/SRS/{year:4d}{month:2d}{day:2d}SRS.txt'
+        extractor2 = '{}/warehouse/{year:4d}/{}'
+        matchdict = {'Instrument': ['SOON'], 'Source': ['SWPC'], 'Provider': ['NOAA'], 'Physobs': ['SRS']}
+        metalist = []
+        for elem in args:
+            if isinstance(elem, a.Time):
+                timerange = TimeRange(elem.start, elem.end)
```

Comment:
Can you not replace this with a call to `pre_hook`?

Reply:
Done